### ISSUE TITLE
Bug: [eqeqeq] npx eslint . did not working after setting up.

### ISSUE THREAD

**Issue by ramwin** [typescript-eslint/typescript-eslint#8683]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.2&fileType=.ts&code=FAehAIAsFMCdvNAHgSwM4Bc3jQewLYJyy6zgCS4GkuArgOaQaJoA2KAds2jbawCbh%2B6AMa4AbnGDIADqWYAzWhxEYUuDuHwBPAIb9%2BACiQAucB1r4ARnACUZi9amz54JSrUateg8bOZYTnp7HAxAjnpgUAgMXFwtXQ5tcCtWRIBrcHYOaDQo-KiXWEVlVXVNHX0jU1Dw%2BnAAH3NLG1hbcABvYHBwMHBaNAQAXhHwTkxofXBcBXAR7t6IDVZkjQQ0GV0RBF0SZUF5nr6rWC306GYOnF4Bc1xuDF3mXXNoAHcszmgFlFnDDG0MmgM3ASDmQ3AAHJHK1IbYFl0ej14BhaLBNGCANTgACMAG4FgBfH5-AFAkFgkYQyEBIJwzoLZEXNEY8DYyE4yEEnrEnrUEgfHIfACisBIsEMACI3iQIlRAdBJbYCcSgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
// here exists some error I thought eslint should discover
export function myadd(x: number): number
export function myadd(x: string): string

// too many blank lines






export function myadd(x: string | number) {
    // use === instead of ==
    // only one space around ==
    // bracket { should not start a new line
    if (typeof x ==    'number')
    {
        return x + 1;
    }
    if (typeof x === 'string') {
        return x + '1';
    }
    throw new Error("wrong type");
}
```

### ESLint Config

```javascript
// @ts-check

import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  eslint.configs.recommended,
  ...tseslint.configs.recommended,
);
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    "target": "ES2020",
    "sourceMap": true,
    "declaration": true,
    "outDir": "dist"
  },
  "include": [
    "src/**/*"
  ]
}
```

### Expected Result

After I run `npx eslint .`,
the `==` in code will be replaced by `===`

### Actual Result

There was no change on the code.

### Additional Info

I followed the [setup documentation](https://typescript-eslint.io/getting-started), it says

> This code will enable our [recommended configuration](https://typescript-eslint.io/users/configs) for linting.

So I'm wonderting if something went wrong, or it was because the rule eqeqeq was not included in the default configuration.

I create a branch to reproduce it, you can try it by:
```bash
git clone git@github.com:ramwin/ramwints.git -b flat-eslint
cd ramwints
npm install && npx eslint .

git status 
# nothing changed
```

**Comment by bradzacher** [comment]
You need to pass the `--fix` flag to run the fixers.
ESLint does not run the by default.

**Comment by bradzacher** [comment]
Additionally we don't use issues as a support forum, as per our contributing guide.

https://typescript-eslint.io/contributing/issues#questions-and-support-requests

The contributing guide includes alternatives should you want support.

In particular here if you are looking for support with using eslint itself - consider talking to the ESLint project https://eslint.org

**Comment by ramwin** [comment]
> You need to pass the `--fix` flag to run the fixers. ESLint does not run the by default.

I ran the command `npx eslint . --fix` and the problem still exists.

**Comment by ramwin** [comment]
> Additionally we don't use issues as a support forum, as per our contributing guide.
> 
> https://typescript-eslint.io/contributing/issues#questions-and-support-requests
> 
> The contributing guide includes alternatives should you want support.
> 
> In particular here if you are looking for support with using eslint itself - consider talking to the ESLint project https://eslint.org

I followed the Quickstart documentation and found it didn't work. There should be some bug or misleading in the documentations. So I thought it was an issue.

**Comment by bradzacher** [comment]
The eqeqeq rule is not part of the recommended sets. So it's not fixing anything because it's not turned on.
Check out the ESLint docs for more info on enabling rules.

**Comment by ramwin** [comment]
I found where the problems is. The default eslint version in npm.js 8.57.0, it didn't support flat config system. But the documentation use the flat config format by default. So it didn't work.  

I recommend setting the legacy format document as default in quickstart until the eslint officially publish the v9.0.0 version.

**Comment by bradzacher** [comment]
That is incorrect.
ESLint will use the flat config file by default in v8 if there is no legacy config in the folder.
